### PR TITLE
feat: Add affinity

### PR DESCRIPTION
**Pull request by averyyip**
…erce

## Description

_Detailed description of the changes and the intent behind them_

## Checklist

- [ ] Changes are covered by tests if behavior has been changed or added
- [ ] Tests have 100% coverage
- [ ] If code changes were made, the version in `package.json` has been bumped (matching semver)

## Resolves

If the PR resolves an open issue tag it here. For example: `Resolves #34`

### CODE REVIEW

File: src/platforms/affinity/client.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  makeRequestFactory,
+} from '@/sdk/client';
+
+const request = makeRequestFactory(async (auth, options) => {
+  return {
+    ...options,
+    url: formatUrl(`https://api.affinity.co`, options.url),
+    headers: {
+      ...options.headers,
+      Authorization: formatBasicAuthHeader('', await auth.getToken()),
```
Comment (averyyip) [inline src/platforms/affinity/client.ts:13]: Affinity is that weird...

File: src/platforms/affinity/index.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { client } from '@/platforms/affinity/client';
+import { icon } from '@/platforms/affinity/icon';
+import { auth, platform } from '@/sdk';
+
+export default platform('affinity', {
+  auth: auth.basic({
+    questions: [{ type: 'text', id: 'password', label: 'API Token' }],
+  }),
```
Comment (rayepps) [inline src/platforms/affinity/index.ts:8]: should the label for this be API token?

File: src/platforms/shopify/index.ts [hunk 3]
```diff
@@ -3,8 +3,6 @@ import * as constants from '@/platforms/shopify/constants';
 import { icon } from '@/platforms/shopify/icon';
 import { auth, platform } from '@/sdk';
 
-export * as types from '@/platforms/slack/schemas';
-
```
Comment (rayepps) [inline src/platforms/shopify/index.ts:7]: is this intentional?

File: src/platforms/affinity/index.ts (hunk 2 above)
Comment (averyyip) [reply src/platforms/affinity/index.ts:8]: nope. The funny thing you'll start to see is that there are quite a few integrations that uses api token in basic auth (username:password + base64 encode

File: src/platforms/affinity/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { client } from '@/platforms/affinity/client';
+import { icon } from '@/platforms/affinity/icon';
+import { auth, platform } from '@/sdk';
+
+export default platform('affinity', {
+  auth: auth.basic({
+    questions: [{ type: 'text', id: 'password', label: 'API Token' }],
```
Comment (rayepps) [inline src/platforms/affinity/index.ts:7]: Don't we need both a username and a password question for basic auth? I'm looking at the basic auth PR, it looks like it needs an answer at https://github.com/vesselapi/integrations/pull/157/files#diff-0883450c0459616d32193a78474a5358d92fcc1785ad4820ecba2d23792b926cR130

Comment (averyyip) [reply src/platforms/affinity/index.ts:7]: so the funny thing is that some of them don't require `username` or `password`

Comment (averyyip) [reply src/platforms/affinity/index.ts:7]: Hence the `??`

File: src/platforms/affinity/index.ts (hunk 2 above)
Comment (rayepps) [reply src/platforms/affinity/index.ts:8]: Oh I meant the `label` attribute on line 7 where it says `'API Token'`, makes sense with what you said

File: src/platforms/affinity/index.ts (hunk 4 above)
Comment (rayepps) [reply src/platforms/affinity/index.ts:7]: ehhhhhh say what? Like the basic auth is literally `base64('password:')` where password is the api token?

Comment (averyyip) [reply src/platforms/affinity/index.ts:7]: Yup, I've seen both 
```
base64(`${API token}:`}

and 

base64(`:${API token}`}
```

**Review by rayepps — APPROVED** [review]

**Pull request merged**